envoy/api/v2/core/BUILD

```diff
@@ -0,0 +1,76 @@
+load("//bazel:api_build_system.bzl", "api_proto_library", "api_go_proto_library", "api_go_grpc_library")
+
+licenses(["notice"])  # Apache 2
+
+api_proto_library(
+    name = "address",
+    srcs = ["address.proto"],
+)
+
+api_go_proto_library(
+    name = "address",
+    proto = ":address",
+)
+
+api_proto_library(
+    name = "base",
+    srcs = ["base.proto"],
+)
+
+api_go_proto_library(
+    name = "base",
+    proto = ":base",
+    deps = [":address_go_proto"],
+)
+
+api_proto_library(
+    name = "health_check",
+    srcs = ["health_check.proto"],
+    deps = [":base"],
+)
+
+api_go_proto_library(
+    name = "health_check",
+    proto = ":health_check",
+    deps = [":base_go_proto"],
+)
+
+api_proto_library(
+    name = "config_source",
+    srcs = ["config_source.proto"],
+    deps = [
+        ":base",
+        ":grpc_service",
+    ],
```

Comment:
Are you still planning on adding visibility restrictions? I'd like us to do something to reduce the likelihood of future churn happening again; it can be visibility, a Go build test or even some good local build validation that we can add to the PR that can assure us we won't churn further.

Reply:
I can add visibility, but that requires that dependencies must be proto packages, not individual protos. Go rules is harder to achieve, that amounts to writing all go rules again, and that was hard before and is still (given that @misterwilliam did only partial subset of protos).

Reply:
Surprisingly, `proto_library` actually has no visibility attribute.

Reply:
PTAL, added "friends" package groups for visibility constraints.